src/stirling/source_connectors/socket_tracer/socket_trace_connector.cc

```diff
@@ -73,6 +73,16 @@ DEFINE_string(socket_trace_data_events_output_path, "",
               "binary format; otherwise, text format.");
 
 // PROTOCOL_LIST: Requires update on new protocols.
+//
+// The infer_protocol function does not work on kernels
+// older than 5.2 when all protocols are enabled. Since
+// the TraceMode::OnForNewerKernel enum results in a
+// 'truthy' define, TraceMode::OnForNewerKernel and
+// TraceMode::Off must be munged into 0 in order for the
+// compiler to remove dead conditional statements resulting
+// in a BPF program within the old program limit (< 4096 insns).
+// This means the TransferSpec's enabled field must be used
+// after SocketTraceConnector::EnableIfNeeded is called.
```

Comment:
Sorry to nit. Still want to tweak this a bit, since the piece about TransferSpec has less context here. How about something simpler like:

// Due to BPF instruction limits (< 4096 instructions) on kernels older than 5.2,
// we can't simultaneously enable all protocols. Thus, some protocols
// are only enabled on newer kernels.

Reply:
No worries. I thought there may have been some missing context on your [first comment](https://github.com/pixie-io/pixie/pull/603#discussion_r984026335), but hopefully this is good now!